tests/UnifiedSpecTests/Util.php

```diff
@@ -26,13 +34,113 @@
+            'assertSessionTransactionState' => ['session', 'state'],
+            'assertSessionUnpinned' => ['session'],
+            'failPoint' => ['client', 'failPoint'],
+            'targetedFailPoint' => ['session', 'failPoint'],
+            'loop' => ['operations', 'storeErrorsAsEntity', 'storeFailuresAsEntity', 'storeSuccessesAsEntity', 'storeIterationsAsEntity'],
+        ],
+        Client::class => [
+            'createChangeStream' => ['pipeline', 'typeMap', 'session', 'fullDocument', 'resumeAfter', 'startAfter', 'startAtOperationTime', 'batchSize', 'collation', 'maxAwaitTimeMS'],
+            'listDatabaseNames' => ['authorizedDatabases', 'filter', 'maxTimeMS', 'session'],
+            'listDatabases' => ['authorizedDatabases', 'filter', 'maxTimeMS', 'session'],
+        ],
+        Database::class => [
+            'aggregate' => ['pipeline', 'typeMap', 'session', 'useCursor', 'allowDiskUse', 'batchSize', 'bypassDocumentValidation', 'collation', 'comment', 'explain', 'hint', 'let', 'maxAwaitTimeMS', 'maxTimeMS'],
+            'createChangeStream' => ['pipeline', 'typeMap', 'session', 'fullDocument', 'resumeAfter', 'startAfter', 'startAtOperationTime', 'batchSize', 'collation', 'maxAwaitTimeMS'],
+            'createCollection' => ['collection', 'typeMap', 'session', 'autoIndexId', 'capped', 'collation', 'expireAfterSeconds', 'flags', 'indexOptionDefaults', 'max', 'maxTimeMS', 'size', 'storageEngine', 'timeseries', 'validationAction', 'validationLevel', 'validator'],
+            'dropCollection' => ['collection', 'typeMap', 'session'],
+            'listCollectionNames' => ['authorizedCollections', 'filter', 'maxTimeMS', 'session'],
+            'listCollections' => ['authorizedCollections', 'filter', 'maxTimeMS', 'session'],
+            'runCommand' => [
+                'command', 'typeMap', 'session',
+                'commandName', // Argument is not used in PHPLib
+            ],
+        ],
+        Collection::class => [
+            'aggregate' => ['pipeline', 'typeMap', 'session', 'useCursor', 'allowDiskUse', 'batchSize', 'bypassDocumentValidation', 'collation', 'comment', 'explain', 'hint', 'let', 'maxAwaitTimeMS', 'maxTimeMS'],
+            'bulkWrite' => ['requests', 'session', 'ordered', 'bypassDocumentValidation'],
+            'createChangeStream' => ['pipeline', 'typeMap', 'session', 'fullDocument', 'resumeAfter', 'startAfter', 'startAtOperationTime', 'batchSize', 'collation', 'maxAwaitTimeMS'],
+            'createFindCursor' => ['filter', 'typeMap', 'session', 'allowDiskUse', 'allowPartialResults', 'batchSize', 'collation', 'comment', 'cursorType', 'hint', 'limit', 'max', 'maxAwaitTimeMS', 'maxScan', 'maxTimeMS', 'min', 'modifiers', 'noCursorTimeout', 'oplogReplay', 'projection', 'returnKey', 'showRecordId', 'skip', 'snapshot', 'sort'],
+            'createIndex' => [
+                'keys', 'commitQuorum', 'maxTimeMS', 'session',
+                'name', // Argument is not used in PHPLib
+            ],
```

Comment:
```suggestion
            'createIndex' => ['keys', 'commitQuorum', 'maxTimeMS', 'name', 'session'],
```

I misspoke when I said that `name` was not used. It is a valid option and will be respected by PHP if set. When not set, the name is generated from the `keys` option.

This is documented in [`MongoDB\Collection::createIndex()`](https://docs.mongodb.com/php-library/current/reference/method/MongoDBCollection-createIndex/), which is also referenced by [`MongoDB\Collection::createIndexes()`](https://docs.mongodb.com/php-library/current/reference/method/MongoDBCollection-createIndexes/#indexes-parameter).

In the code, you'll find that both `createIndex()` and `createIndexes()` delegate to the CreateIndexes operation. That, in turn, uses the IndexInput class to validate index options. The specific code where `name` is generated if unset can be found [here](https://github.com/mongodb/mongo-php-library/blob/1.11.0/src/Model/IndexInput.php#L66).

Note: index creation is a rather unique case where we have a separate class to validate input arguments, in contrast to other operation classes that handle all of their validation in the constructor (including BulkWrite, which is quite complex).